File: app/src/main/java/com/letianpai/robot/time/callback/CustomClockViewUpdateCallback.java
package com.letianpai.robot.time.callback;

import com.letianpai.robot.time.parser.custom.CustomWatchConfig;

/**
 * @author liujunbin
 */
public class CustomClockViewUpdateCallback {

    private CustomClockViewUpdateListener mCustomClockViewUpdateListener;

    private static class GeneralInfoCallbackHolder {
        private static CustomClockViewUpdateCallback instance = new CustomClockViewUpdateCallback();
    }

    private CustomClockViewUpdateCallback() {

    }

    public static CustomClockViewUpdateCallback getInstance() {
        return GeneralInfoCallbackHolder.instance;
    }

    public interface CustomClockViewUpdateListener {
        void onCustomClockViewChanged(CustomWatchConfig customWatchConfig);

    }

    public void setCustomClockViewUpdateListener(CustomClockViewUpdateListener listener) {
        this.mCustomClockViewUpdateListener = listener;
    }

    public void setCustomClockInfo(CustomWatchConfig customWatchConfig) {
        if (mCustomClockViewUpdateListener != null) {
            mCustomClockViewUpdateListener.onCustomClockViewChanged(customWatchConfig);
        }
    }


}

File: app/src/main/java/com/letianpai/robot/time/callback/CustomClockViewUpdateCallback.kt
package com.letianpai.robot.time.callback

import com.letianpai.robot.time.parser.custom.CustomWatchConfig

/**
 * @author liujunbin
 */
class CustomClockViewUpdateCallback private constructor() {
    private var mCustomClockViewUpdateListener: CustomClockViewUpdateListener? = null

    private object GeneralInfoCallbackHolder {
        val instance = CustomClockViewUpdateCallback()
    }

    fun interface CustomClockViewUpdateListener {
        fun onCustomClockViewChanged(customWatchConfig: CustomWatchConfig)
    }

    fun setCustomClockViewUpdateListener(listener: CustomClockViewUpdateListener?) {
        this.mCustomClockViewUpdateListener = listener
    }

    fun setCustomClockInfo(customWatchConfig: CustomWatchConfig) {
        if (mCustomClockViewUpdateListener != null) {
            mCustomClockViewUpdateListener!!.onCustomClockViewChanged(customWatchConfig)
        }
    }


    companion object {
        val instance: CustomClockViewUpdateCallback
            get() = GeneralInfoCallbackHolder.instance
    }
}
